riscv/triggers.cc

```diff
@@ -261,12 +268,8 @@ void itrigger_t::tdata1_write(processor_t * const proc, const reg_t val, const b
 
 std::optional<match_result_t> itrigger_t::detect_trap_match(processor_t * const proc, const trap_t& t) noexcept
```

Comment:
Please file another PR to remove the copypasta between `itrigger_t::detect_trap_match()` and `etrigger_t::detect_trap_match()`

Reply:
We can move those copypasta to ```module_t::detect_trap_match()``` and update the function of ```itrigger_t``` and ```etrigger_t``` to 
```
... itrigger_t::detect_trap_match(bool interrupt, reg_t bit)
{
  assert(bit < xlen);
  if (interrupt && ((bit == 0 && nmi) || ((tdata2 >> bit) & 1))) {
    hit = true;
    return match_result_t(TIMING_AFTER, action);
  }
  return std::nullopt;
}
... etrigger_t::detect_trap_match(bool interrupt, reg_t bit)
{
  assert(bit < xlen);
  if (!interrupt && ((tdata2 >> bit) & 1)) {
    hit = true;
    return match_result_t(TIMING_AFTER, action);
  }
  return std::nullopt;
}
```
How do you feel about this?

Reply:
Still too much copypasta. The only difference here is the condition in the `if()`, so that should be a virtual function returning a bool. A parent class (`trap_common_t`?) has the `detect_trap_match()` method that calls this virtual function.

But again, new PR.

Reply:
OK will do

Reply:
Add trap_common_t in PR (https://github.com/riscv-software-src/riscv-isa-sim/pull/1207).